src/util.c

```diff
@@ -487,55 +489,88 @@ systemd_escape(char* string)
 gboolean
 netplan_delete_connection(const char* id, const char* rootdir)
 {
-    g_autofree gchar* del = NULL;
+    g_autofree gchar* yaml_path = NULL;
     g_autoptr(GError) error = NULL;
     NetplanNetDefinition* nd = NULL;
+    FILE* patch = NULL;
     gboolean ret = TRUE;
 
-    NetplanState* np_state = netplan_state_new();
-    NetplanParser* npp = netplan_parser_new();
+    NetplanParser* input_parser = netplan_parser_new();
+    NetplanState* input_state = netplan_state_new();
+    NetplanParser* output_parser = NULL;
+    NetplanState* output_state = NULL;
 
     /* parse all YAML files */
-    if (   !netplan_parser_load_yaml_hierarchy(npp, rootdir, &error)
-        || !netplan_state_import_parser_results(np_state, npp, &error)) {
+    if (   !netplan_parser_load_yaml_hierarchy(input_parser, rootdir, &error)
+        || !netplan_state_import_parser_results(input_state, input_parser, &error)) {
+        g_warning("netplan_delete_connection: Cannot parse input: %s\n", error->message);
+        ret = FALSE;
+        goto cleanup;
+    }
+
+    /* find specified netdef in input state */
+    nd = netplan_state_get_netdef(input_state, id);
+    if (!nd) {
+        g_fprintf(stderr, "netplan_delete_connection: Cannot delete %s, does not exist.", id);
+        ret = FALSE;
+        goto cleanup;
+    }
+
+    /* Build up a tab-separated YAML path for this Netdef (e.g. network.ethernets.eth0=...) */
+    yaml_path = g_strdup_printf("network\t%s\t%s", netplan_def_type_name(nd->type), id);
+
+    /* create a temporary file in memory, to hold our YAML patch */
+    patch = tmpfile();
+    int patch_fd = fileno(patch);
+    if (!netplan_util_create_yaml_patch(yaml_path, "NULL", patch_fd, &error)) {
         // LCOV_EXCL_START
-        g_fprintf(stderr, "%s\n", error->message);
+        g_fprintf(stderr, "netplan_delete_connection: Cannot create YAML patch: %s", error->message);
         ret = FALSE;
         goto cleanup;
         // LCOV_EXCL_STOP
     }
+    fflush(patch);
```

Comment:
I don't know if I would feel comfortable using C streams (`FILE *`) operations after passing around the underlying fd to a function.

But if you have experience with it, go for it :)

Otherwise, I would recommend dropping fflush() and using lseek() instead of rewind()

Reply:
Is the fflush() really needed here? I think that at this point there is nothing in the stream buffer as the writer only used the file descriptor.

Reply:
ACK. `netplan_util_create_yaml_patch()` is actually flushing the FD itself, so no need to do it here.